astropy/io/misc/tests/test_yaml.py

```diff
@@ -316,3 +316,13 @@ def test_yaml_load_of_object_arrays_fail():
             order: C
             shape: !!python/tuple [3]"""
         )
+
+
+def test_serialize_mixin_column(tmp_path):
+    # see https://github.com/astropy/astropy/issues/16414
```

Comment:
I'm confused why our regular tests didn't catch this. We test a whole slew of different `SkyCoord` for all serializations (see `astropy/io/tests/mixin_columns.py`) - was the problem specific to `frame="galactic"`? If so, perhaps add an extra `SkyCoord` in `mixin_columns`? (Though note that one then needs to add quite a bit of detail further down in the file...) Though in a way also fine just as is - after all, it is the incantation that proved there was a problem.

Reply:
The problem is actually *not* specific to `frame="galactic"`, so I'll drop this argument to avoid confusion.
According to its docstring, `astropy/io/tests/mixin_columns.py` is used in `ascii/tests/test_ecsv.py`, `fits/tests/test_connect.py`, and `misc/tests/test_hdf5.py`, but not in yaml tests (yet). I'll see if there are any problems with using it here too.

Reply:
I've wrote a new test based on `test_hdf5.py::test_hdf5_mixins_qtable_to_table`, which also fails on main and pass with this branch. I ditched the first test, let me know if I should keep both.